scipy/signal/_signaltools.py

```diff
@@ -4841,6 +4848,8 @@ def filtfilt(b, a, x, axis=-1, padtype='odd', padlen=None, method='pad',
     if edge > 0:
         # Slice the actual signal from the extended signal.
         y = axis_slice(y, start=edge, stop=-edge, axis=axis)
+        if is_torch(xp):
+            y = y.copy()    #  pytorch/pytorch#59786 : no negative strides in pytorch
```

Comment:
It's unclear to me why this is needed; the slicing is done in `axis_slice` above, once we get here the function is done already.

Reply:
There's an `axis_revese` just above which leads to

```
4848 	    # Reverse y.
4849 	    y = axis_reverse(y, axis=axis)
4850 	
4851 ->	    if edge > 0:
4852 	        # Slice the actual signal from the extended signal.
4853 	        y = axis_slice(y, start=edge, stop=-edge, axis=axis)
4854 	    #    if is_torch(xp):
4855 	    #        y = y.copy()    #  pytorch/pytorch#59786 : no negative strides in pytorch
4856 	
(Pdb) p y.strides
(-8,)
```
and then `xp.asarray(y)` fails if `xp` is `torch`.

Reply:
`xp.flip` should be functional, it won't modify in-place for any library (that would be a serious bug). So maybe a useful cleanup for a future PR?

Reply:
Ah, then it does not get any simpler than what's here now. `axis_reverse` does things without copying, basically a `slice(None, None, -1)`, and it's only torch which requires an extra copy. One could stuff a `.copy` into `axis_reverse`  but then it returns a copy or a view depending on a backend. Which is a no-no if we can help it, even for private helpers.

Reply:
I didn't say that `flip` has to copy! That would be bad indeed. It's allowed to return a view, that is still functional.

```python
>>> x = np.arange(100_000)
>>> %timeit x.copy()
33.2 μs ± 1.48 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
>>> %timeit np.flip(x)
426 ns ± 3.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

`flip` should be doing exactly what you want here.

Reply:
Minor issue of semantics: you asked about "is in-place", which to me means modifying existing array data (an in-place operation), rather than "can it return a view".